flexmeasures/data/models/planning/storage.py

```diff
@@ -221,25 +221,69 @@ def _prepare(self, skip_validation: bool = False) -> tuple:  # noqa: C901
             StorageScheduler.COLUMNS, start, end, resolution
         )
 
-        ems_power_capacity_in_mw = self.flex_context.get(
-            "ems_power_capacity_in_mw",
-            self.sensor.generic_asset.get_attribute("capacity_in_mw", None),
+        capacity_in_mw = self.sensor.generic_asset.get_attribute("capacity_in_mw", None)
+
+        """
+        Priority order to fetch the site consumption power capacity:
+
+        "site-consumption-capacity" (flex-context) -> "site-power-capacity" (flex-context) ->
+        "consumption_capacity_in_mw" (asset attribute) -> "capacity_in_mw" (asset attribute)
```

Comment:
While editing notation.rst I realised we could be more consistent, and should avoid potentially weird situations in which e.g. the consumption capacity exceeds the power capacity.

Instead of chaining defaults like this, we should only fall back on the corresponding asset attribute, and then set the consumption capacity to the minimum of the power capacity and the consumption capacity. Similarly for the production capacity.

It might help to think of the related concepts as:

- connection capacity (e.g. 2 MVA) -> site-power-capacity
- contacted transport capacity for consumption (e.g. 1.5 MW) -> site-consumption-capacity
- contacted transport capacity for production (e.g. 800 kW) -> site-production-capacity

Reply:
True! I've updated the code to reflect this.